### ISSUE TITLE
The animated progress bar in status bar does not behave as expected 

### ISSUE THREAD

**Issue by merks** [eclipse-platform/eclipse.platform.ui#327]
For me the progress bar is pretty much always jittery never progressing toward completion but rather always restarting after a little flash of green.  Something is badly wrong here.

![image](https://user-images.githubusercontent.com/208716/188895119-ff59050d-9270-4559-b53b-28537da3ff61.png)

![image](https://user-images.githubusercontent.com/208716/188895179-ca6d4e9c-7665-4af0-ad54-809504bb694b.png)

One would expect the bar to represent the % value we see...

I expect that doing a clean build in any workspace will reveal the problem.  Certainly in the Oomphed Eclipse SDK IDE that's the case.

**Comment by jukzi** [comment]
i can not reproduce it in my oomphed workspace:
![image](https://user-images.githubusercontent.com/51790620/189474717-7194bce1-9577-4e95-9ec8-7e52606c253c.png)
would you please paste a screenshot of your progress view at that time? since the number in the progress bar is the *average* of all processes shown in the progress view

**Comment by iloveeclipse** [comment]
On Windows I saw exact same behavior like Ed. Just usual, latest nightly SDK build with usual tasks (build, clean build) over many SDK projects. Looks really ugly, I haven't seen that on Linux, I believe it's Windows only.

Note: I've enabled "Show system jobs" , may be that's the difference.

**Comment by merks** [comment]
Trying to capture new images I see the progress in this strange state:

![image](https://user-images.githubusercontent.com/208716/189516083-151e89ad-3773-4d19-a285-b3ea2163978a.png)

This cannot be canceled and cleaning a project leaves it in that state:

![image](https://user-images.githubusercontent.com/208716/189516134-56bee8f2-c4d5-4be6-8042-fe07770e3ded.png)

Here are some captures for a clean build after a restart::

![image](https://user-images.githubusercontent.com/208716/189516215-2310158b-faa5-400e-ae2f-a470217ee159.png)

![image](https://user-images.githubusercontent.com/208716/189516228-2099a70c-ae6d-41c0-805d-1bd4415e08f3.png)

![image](https://user-images.githubusercontent.com/208716/189516241-1afaf97c-baf9-48b6-9306-2aa0b6bf9232.png)

**Comment by jukzi** [comment]
@merks  does the issue go away if you disable PDE's API analysis?
i guess it's rather a problem of repeated new API Analysis Jobs - which does not give sum up to a meaningfull  total progress

**Comment by merks** [comment]
How do I disable that?  I also see this behavior in my Oomph IDE and there I have no API baseline set...

**Comment by jukzi** [comment]
![image](https://user-images.githubusercontent.com/51790620/190446356-99acf019-c51a-4760-a41e-eae4a74f3ebe.png)

**Comment by merks** [comment]
No, that makes no difference. I don't even see new jobs coming and going while this is happening...

**Comment by iloveeclipse** [comment]
> I haven't seen that on Linux, I believe it's Windows only.

Actually I observe that on Linux too, but it is not as "nervous" as on Windows, where it is really extreme.

**Comment by JanMosigItemis** [comment]
Hi there,

just wanted to report that since I updated to 2022-09 (4.25), I am experiencing the same issue with the progress bar in the status bar on the right lower corner of the IDE window.

When doing background work, it always toggles between like one pixel width and completely empty. Overall IDE behavior is normal, i. e. no jobs fail.

System is:

* Eclipse 4.25
* Running with external JDK17 (17.0.4.1)
* Windows 10.0.19043.2075
* Theming enabled
* Theme: Classic
* Color and font theme: Classic Theme
* Use round tabs: true
* Use mixed fonts and colors for labels: true
* Intel UHD Graphics + Nvidia Quadro T1000

**Comment by jukzi** [comment]
Today i did not see "building" in status bar and a wrong progress - i used fresh oomphed platform + jdk19
![image](https://user-images.githubusercontent.com/51790620/194696733-d44ee68f-4c83-4d38-83f1-46fab7d42b2f.png)
However i only see back and forth of the progress when API analysis jobs spawn. and that is expected as the progress shows a average of the jobs - we could discuss that algorithm if you users agree they want a graphical representation of the longes running job only.

**Comment by merks** [comment]
I think it will look really bad if we ship this behavior given that Windows is the biggest user base.  And this just always happens even when there is just a single job running.  I'm not sure I can find time to fix it with the current SimRel chaos.  😢

**Comment by jukzi** [comment]
Given that it was also seen in 4.25 (https://github.com/eclipse-platform/eclipse.platform.ui/issues/327#issuecomment-1271565710) it can't hurt that much. I would like to fix this, but still can't see that problem on my computers.

**Comment by jukzi** [comment]
would be great if someone with the problem could at least debug it:
https://github.com/eclipse-platform/eclipse.platform.ui/blob/master/bundles/org.eclipse.ui.workbench/Eclipse%20UI/org/eclipse/ui/internal/progress/ProgressAnimationItem.java#L224
to see if the worked amount is wrong or it is just a visual defect. I could imaging that it is just a problem that one line earlier the beginTask() resets the position just before setting the expected value. however all available snippets use that pattern.

**Comment by fabioz** [comment]
Happens here too with 4.26 (2022-12) on Windows.

A .gif with the issue:

![eclipse_progress](https://user-images.githubusercontent.com/117621/206852271-73b9d68b-a18d-4b2e-b9c0-4d62acfe4b60.gif)

**Comment by fabioz** [comment]
Just to note, the PR at: https://github.com/eclipse-platform/eclipse.platform.ui/pull/453 does seem to fix it for me.
